evm_loader/program/src/evm/opcode.rs

```diff
@@ -1289,7 +1298,10 @@ impl<B: Database> Machine<B> {
             self,
             super::tracing::Event::BeginVM {
                 context,
-                code: code.to_vec()
+                code: code
+                    .as_deref()
+                    .map(|code| code.to_vec())
+                    .unwrap_or_default(),
```

Comment:
If the code is borrowed from an account, this will clone that data into a newly allocated vec. Is this necessary?